pinot-plugins/pinot-input-format/pinot-clp-log/src/main/java/org/apache/pinot/plugin/inputformat/clplog/CLPLogMessageDecoder.java

```diff
@@ -38,18 +38,27 @@
  * details. The implementation is based on {@link org.apache.pinot.plugin.inputformat.json.JSONMessageDecoder}.
  */
 public class CLPLogMessageDecoder implements StreamMessageDecoder<byte[]> {
+  public static final String ERROR_SAMPLING_PERIOD_CONFIG_KEY = "errorSamplingPeriod";
   private static final Logger LOGGER = LoggerFactory.getLogger(CLPLogMessageDecoder.class);
+  private static final int DEFAULT_ERROR_SAMPLING_PERIOD = 10000;
 
   private RecordExtractor<Map<String, Object>> _recordExtractor;
+  // Period at which errors should be sampled for printing:
+  // * 0            - Don't print errors
+  // * n > 0        - Print one error every n errors
+  private int _errorSamplingPeriod = DEFAULT_ERROR_SAMPLING_PERIOD;
+  private int _numErrorsUntilNextPrint = 0;
 
   @Override
   public void init(Map<String, String> props, Set<String> fieldsToRead, String topicName)
       throws Exception {
     String recordExtractorClass = null;
     String recordExtractorConfigClass = null;
+    String errorSamplingPeriodString = null;
     if (null != props) {
       recordExtractorClass = props.get(RECORD_EXTRACTOR_CONFIG_KEY);
       recordExtractorConfigClass = props.get(RECORD_EXTRACTOR_CONFIG_CONFIG_KEY);
+      errorSamplingPeriodString = props.get(ERROR_SAMPLING_PERIOD_CONFIG_KEY);
```

Comment:
can you please provide an example of the props.

Reply:
```
"streamConfigs": {
    "stream.kafka.decoder.prop.errorSamplingPeriod": "**100000000**",
 },
 ```